src/main/kotlin/io/github/inductiveautomation/kindling/idb/tagconfig/model/ProviderStatistics.kt

```diff
@@ -0,0 +1,135 @@
+package io.github.inductiveautomation.kindling.idb.tagconfig.model
+
+import io.github.inductiveautomation.kindling.utils.splitCamelCase
+import java.util.Locale
+import kotlin.reflect.KProperty
+
+@Suppress("MemberVisibilityCanBePrivate")
+class ProviderStatistics {
+    val orphanedTags = ListStatistic<Node>("orphanedTags")
+
+    val all = listOf(
+        QuantitativeStatistic("totalAtomicTags"),
+        MappedStatistic("dataTypes"),
+        MappedStatistic("valueSources"),
+        QuantitativeStatistic("totalFolders"),
+        QuantitativeStatistic("totalUdtInstances"),
+        QuantitativeStatistic("totalUdtDefinitions"),
+        QuantitativeStatistic("totalTagsWithAlarms"),
+        QuantitativeStatistic("totalAlarms"),
+        QuantitativeStatistic("totalTagsWithHistory"),
+        QuantitativeStatistic("totalTagsWithEnabledScripts"),
+        QuantitativeStatistic("totalEnabledScripts"),
+        DependentStatistic("totalOrphanedTags", orphanedTags) { it.size },
+        QuantitativeStatistic("totalReadOnlyTags")
+    )
+
+    private val statsMap = all.associateBy { it.name }
+
+    val totalAtomicTags: QuantitativeStatistic by statsMap
+    val totalFolders: QuantitativeStatistic by statsMap
+    val totalUdtInstances: QuantitativeStatistic by statsMap
+    val totalTagsWithAlarms: QuantitativeStatistic by statsMap
+    val totalUdtDefinitions: QuantitativeStatistic by statsMap
+    val totalAlarms: QuantitativeStatistic by statsMap
+    val totalTagsWithHistory: QuantitativeStatistic by statsMap
+    val totalTagsWithEnabledScripts: QuantitativeStatistic by statsMap
+    val totalEnabledScripts: QuantitativeStatistic by statsMap
+    val dataTypes: MappedStatistic by statsMap
+    val valueSources: MappedStatistic by statsMap
+    val totalOrphanedTags: DependentStatistic<Int, MutableList<Node>> by statsMap
+    val totalReadOnlyTags: QuantitativeStatistic by statsMap
+
+    fun processNodeForStatistics(node: Node) {
```

Comment:
This is a somewhat half-baked thought as I read through things, so take it with a grain of salt, but...

It might be nice if you could separate these count functions from the loop here. That is, have each statistic say "I care about whether it's a `udtDefinition`", or "I care about whether it's an `atomicTag`" _on themselves_, and then it's much, much easier to add or remove statistics